pkgs/initiator/initiator.go

```diff
@@ -608,115 +620,148 @@ func (c *Initiator) StartDKG(id [24]byte, withdraw []byte, ids []uint64, network
 	depositData, err := json.Marshal(depositDataJson)
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, nil, err
 	}
 	keysharesData, err := json.Marshal(keyshares)
 	if err != nil {
-		return nil, nil, err
+		return nil, nil, nil, err
+	}
+	ceremonySigs, err := c.GetCeremonySigs(dkgResults)
+	if err != nil {
+		return nil, nil, nil, err
+	}
+	ceremonySigsBytes, err := json.Marshal(ceremonySigs)
+	if err != nil {
+		return nil, nil, nil, err
 	}
 	resultMsg := &wire.ResultData{
 		Operators:     ops,
 		Identifier:    id,
 		DepositData:   depositData,
 		KeysharesData: keysharesData,
+		CeremonySigs:  ceremonySigsBytes,
 	}
 	err = c.sendResult(resultMsg, ops, consts.API_RESULTS_URL, id)
 	if err != nil {
-		c.Logger.Error("🤖 Error storing results at operators", zap.Error(err))
+		return nil, nil, nil, fmt.Errorf("🤖 Error storing results at operators %w", err)
 	}
-	return depositDataJson, keyshares, nil
+	return depositDataJson, keyshares, ceremonySigs, nil
 }
 
-func (c *Initiator) StartReshare(newId, oldID [24]byte, oldIDs, newIDs []uint64, owner common.Address, nonce uint64) (*KeyShares, error) {
-
+func (c *Initiator) StartReshare(id [24]byte, newIDs []uint64, keysharesFile []byte, ceremonySigs []byte) (*KeyShares, *CeremonySigs, error) {
+	var ks *KeyShares
+	if err := json.Unmarshal(keysharesFile, &ks); err != nil {
+		return nil, nil, err
+	}
+	var cSigs *CeremonySigs
+	if err := json.Unmarshal(ceremonySigs, &cSigs); err != nil {
+		return nil, nil, err
+	}
+	cSigBytes, err := hex.DecodeString(cSigs.Sigs)
+	if err != nil {
+		return nil, nil, err
+	}
+	oldIDs := ks.Shares[0].Payload.OperatorIDs
+	owner := common.HexToAddress(ks.Shares[0].OwnerAddress)
+	nonce := ks.Shares[0].OwnerNonce
 	oldOps, err := ValidatedOperatorData(oldIDs, c.Operators)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 	newOps, err := ValidatedOperatorData(newIDs, c.Operators)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
-
 	pkBytes, err := crypto.EncodePublicKey(&c.PrivateKey.PublicKey)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
-
-	instanceIDField := zap.String("instance_id", hex.EncodeToString(newId[:]))
+	instanceIDField := zap.String("instance_id", hex.EncodeToString(id[:]))
 	c.Logger.Info("🚀 Starting ReSHARING ceremony", zap.String("initiator_id", string(pkBytes)), zap.Uint64s("old_operator_ids", oldIDs), zap.Uint64s("new_operator_ids", newIDs), instanceIDField)
-
 	// compute threshold (3f+1)
 	oldThreshold := len(oldIDs) - ((len(oldIDs) - 1) / 3)
 	newThreshold := len(newIDs) - ((len(newIDs) - 1) / 3)
-
+	sharesData, err := hex.DecodeString(ks.Shares[0].Payload.SharesData[2:])
+	if err != nil {
+		return nil, nil, err
+	}
 	reshare := &wire.Reshare{
 		OldOperators:       oldOps,
 		NewOperators:       newOps,
 		OldT:               uint64(oldThreshold),
 		NewT:               uint64(newThreshold),
-		OldID:              oldID,
 		Owner:              owner,
 		Nonce:              nonce,
```

Comment:
can the resharing nonce be the same as the previous nonce? doesn't make sense to me. a new nonce should probably be provided, otherwise we won't be able to use this to register the validator to SSV

Reply:
True, good catch.

Reply:
Fixed. Added nonce to resharing as a CLI param.